source/MaterialXShaderGen/ShaderGenerators/Glsl/GlslShaderGenerator.h

```diff
@@ -4,6 +4,50 @@
 #include <MaterialXShaderGen/ShaderGenerator.h>
 #include <MaterialXShaderGen/HwShader.h>
 
+/*
+The GLSL shader generator has a number of predefined variables (inputs and uniforms) with set binding rules.
+When these are used by a shader the application must bind them to the expected data. The following table is
+a listing of the variables with a description of what data they should be bound to.
+
+------------------------------------------------------------------------------------------------------------
+    NAME                                TYPE    BINDING
+------------------------------------------------------------------------------------------------------------
+
+Vertex input variables :
+    i_position                          vec3    Vertex position in object space
+    i_normal                            vec3    Vertex normal in object space
+    i_tangent                           vec3    Vertex tangent in object space
+    i_bitangent                         vec3    Vertex bitangent in object space
+    i_texcoord_N                        vec2    Vertex texture coordinate for the N:th uv set
+    i_color3_N                          vec3    Vertex color for the N:th color set (RGB)
```

Comment:
As mentioned above I think we only need i_color_N.

Reply:
For a case where both color3 and color4 is used in the same shader I think we need to keep these seperate? For example a color3 from color set #1 and a color4 from color set #2.

Reply:
Ok, maybe it would be simpler to always use a vec4 i_color_N then (always 4-component input) and use offset/stride as use mentioned if the data on geometry is less than 4 components? Let's discuss this offline before I make the change.